### PR TITLE
fix(Picker): should reset subsequent columns when a column is changed

### LINKED ISSUE #13365: [Bug Report] picker切换的时候，如果有子元素的时候，子元素都是从0开始的
### 重现链接

https://stackblitz.com/edit/vitejs-vite-ygkxkux3?file=src%2Fmain.js,src%2Fcomponents%2FHelloWorld.vue,src%2FApp.vue&terminal=dev

### Vant 版本

最新的

### 描述一下你遇到的问题。

我提供了最小的demo

### 重现步骤

我提供了最小的demo

### 设备/浏览器

_No response_

### PR DESCRIPTION
**Pull request by JerryWu1234**
… is changed somehow

fixes: #13365

add a new property for resetChildren 
```js

```

Before submitting a pull request, please read the [contributing guide](https://vant-ui.github.io/vant/#/en-US/contribution).

在提交 pull request 之前，请阅读 [贡献指南](https://vant-ui.github.io/vant/#/zh-CN/contribution)。

### CODE REVIEW

**Comment by JerryWu1234** [comment]
@inottn  Please help to review. I done and fixed all fail tests .

**Comment by inottn** [comment]
I don't know if the previous interactions were intentional; if there is an issue, the onChange logic might need to be modified. If I have time, I will see how to adjust it.

**Comment by JerryWu1234** [comment]
> I don't know if the previous interactions were intentional; if there is an issue, the onChange logic might need to be modified. If I have time, I will see how to adjust it.


I definitely sure that it wasn't intentional. and it is a bug

File: packages/vant/src/picker/Picker.tsx [hunk 1]
```diff
@@ -159,11 +161,15 @@ export default defineComponent({
...
-            );
+          // reset when the first argument is changed
+          if (!isOptionExist(options, value, fields.value) || index > 0) {
+            const isEmptyOption = options.length
+              ? options[0][fields.value.value]
+              : undefined;
+            // there is special situation that when `columnsPlaceholder` property passes into Area component
+            if (isEmptyOption !== AREA_EMPTY_CODE) {
+              setValue(index, isEmptyOption);
+            }
```
[the remark this answers is not in the document]
Comment (JerryWu1234) [reply packages/vant/src/picker/Picker.tsx:172]: @inottn 
Is there a method to distinguish this data from Area's `columnsPlaceholder`?

I suppose we could add a mark when executIng the`formatDataForCascade` function.
```js
{
"text: "City",
"value": "000000",
"children":[],
"isPlaceholder": true
}
```

[the remark this answers is not in the document]
Comment (JerryWu1234) [reply packages/vant/src/picker/Picker.tsx:172]: Otherwise, there are no other fields to indicate that it is incorrect when using Area component .

[the remark this answers is not in the document]
Comment (inottn) [reply packages/vant/src/picker/Picker.tsx:172]: Ignoring the Area logic for the moment, the current fix is flawed. The data received in the confirm event doesn't match the view.

[the remark this answers is not in the document]
Comment (JerryWu1234) [reply packages/vant/src/picker/Picker.tsx:172]: > Ignoring the Area logic for the moment, the current fix is flawed. The data received in the confirm event doesn't match the view.

ok you are right. test no cover this case.
I already fixed

[the remark this answers is not in the document]
Comment (JerryWu1234) [reply packages/vant/src/picker/Picker.tsx]: @inottn I enhanced the edge case.

**Comment by JerryWu1234** [comment]
thanks for point out.
now no Area component error, and added edge case.
please help to review it again

File: packages/vant/src/picker/test/index.spec.tsx [hunk 2]
```diff
@@ -440,3 +440,76 @@ test('should render empty slot when options is empty', async () => {
   await wrapper.setProps({ columns: [{ values: ['foo'] }] });
   expect(wrapper.html()).not.toContain('empty content');
 });
+
+test('should reset the second argument when the first argument is changed somehow', async () => {
```
Comment (inottn) [inline packages/vant/src/picker/test/index.spec.tsx:444]: A view-based description provides greater clarity. Also, update the PR title accordingly.
```suggestion
test('should reset subsequent columns to their first item when a column changes', async () => {
```

File: packages/vant/src/picker/Picker.tsx [hunk 3]
```diff
@@ -159,11 +159,16 @@ export default defineComponent({
         // reset values after cascading
         selectedValues.value.forEach((value, index) => {
           const options = currentColumns.value[index];
-          if (!isOptionExist(options, value, fields.value)) {
-            setValue(
-              index,
-              options.length ? options[0][fields.value.value] : undefined,
-            );
+          // reset when the first argument is changed
```
Comment (inottn) [inline packages/vant/src/picker/Picker.tsx:162]: Without context, I'm having trouble understanding the `argument` here. Please either remove this comment or provide sufficient explanation.

**Comment by chenjiahan** [comment]
I prefer to keep the current behavior. If we reset the option it will break the user experience of date picker or time picker.

**Comment by JerryWu1234** [comment]
> I prefer to keep the current behavior. If we reset the option it will break the user experience of date picker or time picker.

ok, if we keep the current behavior. 
Is there another workaround to fix it? 
@chenjiahan

**Comment by JerryWu1234** [comment]
> How about adding a new prop to enable this new behavior? Such as `resetChildren`.

I think the default should be `false` if we add a new prop.

**Comment by JerryWu1234** [comment]
@chenjiahan @inottn I done.

**Comment by JerryWu1234** [comment]
@inottn please help to review.  because my production needs this feature , thanks

**Comment by inottn** [comment]
> @inottn please help to review. because my production needs this feature , thanks

Sorry for the delay. Vant is currently community-driven, and PR reviews might not be prompt, I've been very busy lately, and I apologize again. If needed, you can temporarily resolve issues using methods like pnpm patch.

**Comment by JerryWu1234** [comment]
> > @inottn please help to review. because my production needs this feature , thanks
> 
> Sorry for the delay. Vant is currently community-driven, and PR reviews might not be prompt, I've been very busy lately, and I apologize again. If needed, you can temporarily resolve issues using methods like pnpm patch.

Please don't forget my PR

**Comment by lllomh** [comment]
> > > @inottn please help to review. because my production needs this feature , thanks
> > 
> > 
> > Sorry for the delay. Vant is currently community-driven, and PR reviews might not be prompt, I've been very busy lately, and I apologize again. If needed, you can temporarily resolve issues using methods like pnpm patch.
> 
> Please don't forget my PR

@JerryWu1234  You have been forgotten  ....

**Review by inottn — APPROVED** [review]
参考了其他组件库的设计，Picker 支持两种模式：级联模式和非级联模式。在级联模式下，上下级选项存在关联关系，当上级选项变更时，下级会自动重置为第一项，这种设计是合理且用户体验友好的。而 DatePicker 等组件采用的是非级联模式，因此不会受到这种联动逻辑的影响。

**Comment by JerryWu1234** [comment]
thanks for your approval
